firefox_ui_harness/runners/base.py

```diff
@@ -41,9 +49,43 @@
     'focusmanager.testmode': True,
     'geo.provider.testing': True,
     'javascript.options.showInConsole': True,
+    'marionette.logging': False,
     'security.notification_enable_delay': 0,
     'signon.rememberSignons': False,
     'startup.homepage_welcome_url': 'about:blank',
     'toolkit.startup.max_resumed_crashes': -1,
     'toolkit.telemetry.enabled': False,
 }
+
+
+class FirefoxUITestRunner(BaseMarionetteTestRunner):
+    def __init__(self, **kwargs):
+        BaseMarionetteTestRunner.__init__(self, **kwargs)
+
+        if not self.server_root:
+            self.server_root = firefox_ui_tests.resources
+
+        # Bug 1146847: This needs a refactoring given that our default
+        # preferences are not coming from the command line
+        self.prefs.update(DEFAULT_PREFS)
+
+        if not kwargs.get('e10s'):
+            self.prefs.update({'browser.tabs.remote.autostart': False})
```

Comment:
This still has an undesirable effect if we're running against a nightly that doesn't have e10s on by default, because I'll be able to pass --e10s but not get e10s enabled and otherwise this change has no effect on the set of prefs enabled for tests. I don't think this comment should reference that bug because I think that bug's going to need to be resolved invalid.